lang/test/org/partiql/lang/ast/PathComponentExprTest.kt

```diff
@@ -14,111 +14,100 @@
-                assertFalse(tc.b.equals(tc.a), "b must not equal a")
-
-                // .hashCode() uniqueness is not guaranteed so this assertion might fail problem someday.
-                // however, the odds of this should be extremely low if `.hashCode()` is implemented effectively.
-                assertNotEquals(
-                    tc.a.hashCode(), tc.b.hashCode(),
-                    "a.hashCode() must not equal b.hashCode(), most likely."
-                )
-            }
-        }
-    }
-
-    fun parametersForEquivalenceTest() =
-        listOf(
-            // is equivalent to itself
-            TestCase(fooSensitive, fooSensitive, true),
-            TestCase(fooInsensitive, fooInsensitive, true),
-            TestCase(oneSensitive, oneSensitive, true),
-            TestCase(oneInsensitive, oneInsensitive, true),
-
-            // is equivalent to a copy of itself
-            TestCase(fooSensitive, fooSensitive.copy(), true),
-            TestCase(fooInsensitive, fooInsensitive.copy(), true),
-            TestCase(oneSensitive, oneSensitive.copy(), true),
-            TestCase(oneInsensitive, oneInsensitive.copy(), true),
-
-            // sensitive PathComponentExpr with a literal string is not equivalent to varying cases
-            TestCase(fooSensitive, fooSensitive.copy(expr = litString("FOO")), false),
-            TestCase(fooSensitive, fooSensitive.copy(expr = litString("FoO")), false),
-            TestCase(fooSensitive, fooSensitive.copy(expr = litString("fOo")), false),
-
-            // insensitive PathComponentExpr with a literal string is equivalent to varying cases
-            TestCase(fooInsensitive, fooInsensitive.copy(expr = litString("FOO")), true),
-            TestCase(fooInsensitive, fooInsensitive.copy(expr = litString("FoO")), true),
-            TestCase(fooInsensitive, fooInsensitive.copy(expr = litString("fOo")), true),
-
-            // isn't equivalent because PathComponentExpr.expr doesn't match (but [PathComponentExpr.case] does.)
-            TestCase(fooSensitive, oneSensitive, false),
-            TestCase(fooInsensitive, oneInsensitive, false),
-            TestCase(fooSensitive, fooSensitive.copy(expr = litString("bat")), false),
-            TestCase(fooInsensitive, fooInsensitive.copy(expr = litString("bat")), false),
-            TestCase(oneSensitive, oneSensitive.copy(expr = litInt(2)), false),
-            TestCase(oneInsensitive, oneInsensitive.copy(expr = litInt(2)), false),
-
-            // isn't equivalent because PathComponentExpr.case doesn't match (but [PathComponentExpr.expr] does.)
-            TestCase(fooSensitive, fooInsensitive, false),
-            TestCase(fooSensitive, oneInsensitive, false),
-            TestCase(oneSensitive, oneInsensitive, false)
-        )
-}
+// TODO: This is a test for node equivalence for the deprecated ExprNode.It has not been removed because of this issue: https://github.com/partiql/partiql-lang-kotlin/issues/878
```

Comment:
Did we actually get into a conclusion that equivalence of AST nodes is not useful?

Reply:
If so, we can remove this file

Reply:
It is not used as of now in the library, but I will not say it is not useful. See https://github.com/partiql/partiql-lang-kotlin/issues/878